pyrokinetics/gk_code/GKInputCGYRO.py

```diff
@@ -491,3 +491,14 @@ def set(
             return
 
         self.data = convert_dict(self.data, local_norm.cgyro)
+
+    def get_poincare_fac(self, dpsidr: float, rhostar: float, Lx: float) -> tuple:
+        """get geometrical factors to generate Poincare map"""
+        qinp = self.data["Q"]
+        shat = self.data["S"]
+        rho = self.data["RMIN"]
```

Comment:
Why not take these from a `local_geometry` object. You could take all of this out of the `GKInput` and put it directly in `GKOutputReader`. You can get `Lx` from `numerics too. 

This also avoid any potential normalisation issues and `local_geometry` has a normalisation conventions along with the units

Reply:
Yes, you are right. I did this because at the beginning I was using `bunit` for CGYRO, which is not defined in GS2 and GENE. Then I changed to make it equal for the three code. So this function is not necessary.